ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,931 @@
+    column values from a child dataset.
+
+The full code in the different engines is more complicated than the simplified process outlined above, especially
+when it comes to executing dependencies, but the broad outline is the same.
+
+Appendix
+========
+
+More information on the work done in the code generator to create the workunit can be found in ecl/eclcc/DOCUMENTATION.rst.
+
+The C++ code can be generated as a single C++ file or multiple files.  The system defaults to multiple C++ files, so that
+they can be compiled in parallel (and to avoid problems some compilers have with very large files).  When multipe C++ files are
+generated the metadata classes and workflow classes are generated in the main module, and the activities are generated
+in the files suffixed with a number.  It may be easier to understand the generated code if it is in one place.  In which case,
+compile your query with the option -fspanMultipleCpp=0.  Use -fsaveCppTempFiles to ensure the C++ files are not deleted
+(the C++ files will appear as helpers in the workunit details).
+
+Key types and interfaces from eclhelper.hpp
+-------------------------------------------
+
+IEclProcess
+    The interface that is used by the workflow engine to execute the different workflow items in the generated code.
+
+ThorActivityKind
+    This enumeration contains one entry for each activity supported by the engines.
+
+ICodeContext
+    This interface is implemented by the engine, and provides a mechanism for the generated code to call back into the
+    engine.   For example resolveChildQuery() is used to obtain a reference to a child query that can then be executed later.
+
+IOutputMetaData
+    This interface is used to describe any meta data associated with the data being processed by the queries.
+
+IHThorArg
+    The base interface for defining information about an activity.  Each activity has an associated interface that is derived
+    from this interface.  E.g. each instance of the sort activity will have a helper class implementing IHThorSortArg in
+    the generated query.  There is normally a corresponding base class for each interface in eclhelper_base.hpp that is
+    used by the generated code e.g. CThorSortArg.
+
+ARowBuilder
+    This abstract base class is used by the transform functions to reserve memory for the rows that are created.
+
+IEngineRowAllocator
+    Used by the generated code to allocate rows and rowsets.  Can also be used to release rows (or call the global function rtlReleaseRow()).
+
+IGlobalCodeContext
+    Provides access to functions that cannot be called inside a graph - i.e. can only be called from the global workflow code.  Most functions
+    are related to the internal implementation of particular workflow item types (e.g. persists).
+
+Glossary
+--------
+
+superfile
+    A composite file which allows a collection of files to be treated as a single compound file.
+
+?What else should go here?
```

Comment:
The glossary section is a bit light but I can't thing of anything else that really 'should' be added here. The following is already covered in the document but may be briefly described here too: graph, workunit, Dali, Casandra,